packages/core/src/new-api/types/module.ts

```diff
@@ -376,4 +376,5 @@ export interface IgnitionModule<
   futures: Set<Future>; // Future created in this module — All of them have to be deployed before returning the results. Note that not all of them are in results
   submodules: Set<IgnitionModule>; // Modules used by this module — Note that there's only one instance of each module
   results: IgnitionModuleResultsT; // The futures returned by the callback passed to buildModule
+  getFutures(): Future[];
```

Comment:
Not sure if I'd add this here, as it restrain us a bit from creating other (static, like yml) representations of the modules.

Or maybe I'm overreacting? I don't have a clear mental model of how having logic here and a static representation would work, nor how/why this method is exposed to the users instead of being an internal utility.

Reply:
Ha! This was my one concession to object-orientation - I wanted to `getFutures` on modules just about everywhere. More than happy to move it back to being a utility function.